### ISSUE TITLE
Automatic transfer of multiplayer maps if client does not have it

### ISSUE THREAD

**Issue by godest88** [OpenRA/OpenRA#12439]
Could not find any duplicates of this, so here we go : 
Many other remakes of older rts games uses this feature (as well as more modern ones like Warcraft III). I have been a big fan of Knights and Merchants remake (www.kamremake.com). They did implement this feature which is really nice and makes it less annoying when trying to play a custom map with friends that do not have it (since one of you created it after all). How it would work is basically when a map is selected that a client does not have, the file will be transfered to the client :+1:

**Comment by yaaaaa** [comment]
Good Idea, my little "Commander" can't start to play every time, then he forgot to klick download button.

**Comment by GraionDilach** [comment]
Isn't this the reason behind the Resource Center?

**Comment by godest88** [comment]
@GraionDilach Am i missing something? I have not used Resource center (but its only a website right?), but having this feature built into the game client itself and making it do it in the background kills all extra work for players.

**Comment by GraionDilach** [comment]
I don't play multiplayer so I never could test it, but AFAIK clients can download maps from the Resource Center in the background if the server forces them to do so - by playing them. Some of the servers are even set up to sync their map pool with the REsource Center for this reason.

**Comment by abcdefg30** [comment]
Yeah, a map uploaded to http://resource.openra.net can be downloaded ingame. The host has to select the map and then each client has the option to download it. Does that fulfill your request?

**Comment by godest88** [comment]
@abcdefg30  Not really, its one way to share it with the entire community, which is great. But its not the same thing as i suggested. Which would mean 0 work for both parties (host + clients who join).

**Comment by Phrohdoh** [comment]
When you join a lobby that has selected a map that your system does not have you are presented with a `Download Map` button under the map preview.

Clicking this will do as it suggests and you will be able to play (assuming nothing goes wrong in the download process).

This is, of course, not automatic and must be done before the game is started.

We could add a setting that enables automatic map download.

**Comment by Snurre86** [comment]
But if you have more then one version of the game sharing resources it will make the map incompatible with the other versions? I remember there was some issue about this that made easier to leave it manual and only force you to download the map before pressing the ready button or starting the game if you are admin. So the only problem left is if the admin that is not you start the game before you have pressed ready.

**Comment by godest88** [comment]
More maps makes the game significantly slower? Come on now, that sounds really silly :) If you have several versions of the clients trying to share maps, how does a version stamp on maps not solve this though?

**Comment by Fincer** [comment]
I have also thought this feature would be a good one but never opened a new issue for it.

Anyway, I support the idea. (y)

**Comment by cjshmyr** [comment]
> more maps make the game slower to load.

This was fixed a while ago, and shouldn't anymore.

**Comment by Snurre86** [comment]
Just try it for your self, I did a week ago. Download it and place it in your folder, record the load time before and after. 

resource.openra.net/maps/zip/

**Comment by godest88** [comment]
@Snurre86  I have several new maps, and i have not noticed any slowdown. Or do you mean the actual download speed?

**Comment by pchote** [comment]
You won't notice any difference with several maps.  The problems happen when you get to hundreds of maps (there are over 3000 custom maps available on the resource center).

**Comment by abcdefg30** [comment]
Closing as duplicate of #17023.
